odoo/tools/populate.py

```diff
@@ -65,7 +65,7 @@ def get_field_variation_date(model: Model, field: Field, factor: int, series_ali
 
     def redistribute(value):
         return SQL(
-            "(%(value)s - (%(factor)s - %(series_alias)s) * floor(%(total_days)s/%(factor)s) * interval '1 days')::%(cast_type)s",
+            "(%(value)s - floor((%(factor)s - %(series_alias)s) * (%(total_days)s::float/%(factor)s)) * interval '1 days')::%(cast_type)s",
```

Comment:
Does it make sense to `floor` the value for datetime fields ? AFAIK, `SELECT 0.1 * interval '1 days';` works fine and you don't want to have all your datetime value to have the same hour:minute:second, right ?

Actually I think always removing floor is also ok for date field because of the case done after.

Reply:
Oh so you're actually reviewing this...

Yup you're right. We can drop the `floor`